org.civicrm.reqangular/src/common/services/compuRoute/route-decorator.js

```diff
@@ -0,0 +1,56 @@
+define([
+    'common/angular',
+    'common/modules/compuRoute'
+], function (angular, module) {
+    'use strict';
+
+    module.factory('RouteDecorator', function () {
+        return {
+            /**
+             * Decorator function that provides new functionality either to $stateProvider or $routeProvider
+             * enabling them to declare common resolve across all states/routes using .resolveForAll() method.
+             * @param provider
+             */
+            enableCommonResolve: function (provider) {
```

Comment:
I like the idea of augmenting both `ngRoute` and `ui.router`, but wouldn't it be better to have, instead of a decorator that needs to handle both and that you need to explicitly use, two separate modules that overrides them?

Right now you need to do this:
```js
define([
    // ...
    'common/services/compuRoute/route-decorator'
], function (angular) {
    angular.module('myApp', [
        'ngRoute', // module to augment
        // ...
        'common.compuRoute' // additional module to reference
    ]).config([ /* ... */, function ($routeProvider, decorator,  /* ... */) {
        decorator.enableCommonResolve($routeProvider);  // explicit opt-in
        
        $routeProvider.resolveForAll({ /* ... */ }).when({ /* ... */ })
    }]);
});
```

But if you were to create two modules, say `compuNgRoute` and `compu.ui.route` which would extend respectively the two modules, everything would happen behind the scenes:

```js
define([
    // ...
    'common/module/compu-ng-route'
], function (angular) {
    angular.module('myApp', [
        'compuNgRoute', // takes care of requiring ngRoute, as its a dependency
    ]).config([ /* ... */, function ($routeProvider, decorator,  /* ... */) {
        // resolveForAll is already available on the provider        
        $routeProvider.resolveForAll({ /* ... */ }).when({ /* ... */ })
    }]);
});
```